openapi-converter/openapi_to_clients_schema/src/openapi.rs

```diff
@@ -0,0 +1,408 @@
+    ///
+    pub fn get_schema<'a>(&'a self, r_or_s: &'a ReferenceOr<Schema>) -> anyhow::Result<&'a Schema> {
+        match r_or_s {
+            ReferenceOr::Reference { reference } => self.ref_to_schema(&reference),
+            ReferenceOr::Item(schema) => Ok(&schema),
+        }
+    }
+
+    ///
+    /// Is this type nullable?
+    ///
+    pub fn is_nullable(&self, id: &str) -> bool {
+        self.components
+            .as_ref()
+            .and_then(|c| c.schemas.get(id))
+            .map(|s| match s {
+                ReferenceOr::Reference { reference: _r } => false,
+                ReferenceOr::Item(item) => item.schema_data.nullable
+            })
+            .unwrap_or_default()
+    }
+
+    ///
+    /// Is this a `not` schema?
+    ///
+    pub fn is_not(&self, r_or_s: &ReferenceOr<Schema>) -> bool {
+        if let Ok(schema) = self.get_schema(r_or_s) {
+            if let SchemaKind::Not {..} = schema.schema_kind {
+                return true;
+            }
+        }
+        false
+    }
+
+    ///
+    /// Merge a series of schemas and return a single combined schema that is guaranteed to not be an `Any`.
+    /// Used to resolve `allOf` directives.
+    ///
+    pub fn merge_schemas(&self, schemas: &Vec<ReferenceOr<Schema>>, data: &SchemaData) -> anyhow::Result<Schema> {
+        let mut merged = AnySchema::default();
+        for schema in schemas {
+            merged = self.merge_in_any(merged, schema)?;
+        }
+
+        Ok(Schema {
+            schema_data: data.clone(),
+            schema_kind: any_to_schema(merged)?,
+        })
+    }
+
+    fn merge_in_any(&self, acc: AnySchema, schema: &ReferenceOr<Schema>) -> anyhow::Result<AnySchema> {
+
+        let schema = self.get_schema(schema)?;
+
+        match &schema.schema_kind {
+            SchemaKind::AllOf { all_of } | SchemaKind::Any(AnySchema{ all_of, .. }) if !all_of.is_empty() => {
+                let mut result = acc;
+                for schema in all_of {
+                    result = self.merge_in_any(result, &schema)?;
+                }
+                return Ok(result);
+            },
+            _ => {},
+        }
+
+        let schema = schema_to_any(&schema.schema_kind)?;
+
+        let result = AnySchema {
+            typ: merge_option(acc.typ, schema.typ)?,
+
+            any_of: merge_vec(acc.any_of, schema.any_of),
+            one_of: merge_vec(acc.one_of, schema.one_of),
+            all_of: merge_vec(acc.all_of, schema.all_of),
+            not: merge_option(acc.not, schema.not)?,
+
+            // Structure properties
+            additional_properties: merge_option(acc.additional_properties, schema.additional_properties)?,
+            items: merge_option(acc.items, schema.items)?,
+
+            unique_items: merge_option(acc.unique_items, schema.unique_items)?,
+            properties: Default::default(),
+            max_properties: merge_option(acc.max_properties, schema.max_properties)?,
```

Comment:
Idle thought: `max_properties` and `min_properties` might have special merging behavior even when the specified values aren't identical, maybe some others too? Probably don't have to handle this now until it comes up in a real spec.